File: 0485_max_consecutive_ones/find_max.py
from typing import List


def findMaxConsecutiveOnes(nums: List[int]) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    res = curr = 0
    for num in nums:
        if num == 1:
            curr += 1
            res = max(res, curr)
        else:
            curr = 0
    return res
            

File: 0485_max_consecutive_ones/find_max.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int findMaxConsecutiveOnes(int[] nums) {
        int res = 0, curr = 0;
        for (int num: nums) {
            if (num == 1) {
                curr++;
                res = Math.max(res, curr);
            } else {
                curr = 0;
            }
        }
        return res;
    }
}